bulk_site_creator/views.py

```diff
@@ -0,0 +1,335 @@
+            else f' and department ID {selected_department_id}' if selected_department_id \
+            else ' and no selected department or course group.'
+        logger.debug(f'Retrieving potential course sites for term ID '
+                     f'{selected_term_id}{logging_dept_cg_text}', extra={"sis_account_id": sis_account_id,
+                                                                         "school_id": school_id,
+                                                                         })
+
+        # Retrieve all course instances for the given term_id and account that do not have Canvas course sites
+        # nor are set to be fed into Canvas via the automated feed
+        potential_course_sites_query = get_course_instance_query_set(
+            selected_term_id, sis_account_id
+        ).filter(canvas_course_id__isnull=True,
+                 sync_to_canvas=0,
+                 bulk_processing=0,
+                 term__term_id=selected_term_id)
+
+        # Filter potential_course_sites_query by course group.
+        if selected_course_group_id and selected_course_group_id != '0':
+            potential_course_sites_query = potential_course_sites_query.filter(course__course_group=selected_course_group_id)
+        # Filter potential_course_sites_query by department.
+        elif selected_department_id and selected_department_id != '0':
+            potential_course_sites_query = potential_course_sites_query.filter(course__department=selected_department_id)
+
+    # TODO maybe better to use template tag unless used elsewhere?
+    # TODO cont. this may be included in a summary generation to be displayed in page (see wireframe and Jira ticket)
+    potential_course_site_count = (
+        potential_course_sites_query.count() if potential_course_sites_query else 0
+    )
+
+    logger.debug(f'Retrieved {potential_course_site_count} potential course sites for term '
+                 f'{selected_term_id}{logging_dept_cg_text}', extra={"sis_account_id": sis_account_id,
+                                                                     "school_id": school_id,
+                                                                     })
+
+    context = {
+        "terms": terms,
+        "potential_course_sites": potential_course_sites_query,
+        "potential_site_count": potential_course_site_count,
+        "canvas_site_templates": canvas_site_templates,
+        "departments": departments,
+        "course_groups": course_groups,
+        'selected_term_id': selected_term_id,
+        'selected_course_group_id': selected_course_group_id,
+        'selected_department_id': selected_department_id,
+        'canvas_url': settings.CANVAS_URL,
+    }
+    return render(request, "bulk_site_creator/new_job.html", context=context)
+
+
+@login_required
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["POST"])
+def create_bulk_job(request: HttpRequest) -> Optional[JsonResponse]:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 22 other functions. By comparison, 99% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 272-334 into a separate function.